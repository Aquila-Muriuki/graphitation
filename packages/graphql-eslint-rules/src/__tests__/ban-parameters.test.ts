import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../ban-parameters";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("ban-parameters", rule, {
  valid: [
    {
      code: `
extend type Query {
  messageOfTheDay(date: Date!): String
}`,
      options: [
        {
          bannedParameters: [
            {
              keywords: ["force", "correlationId"],
              description: "42",
            },
          ],
        },
      ],
    },
  ],
  invalid: [
    {
      code: `
extend type Query {
  messageOfTheDay(date: Date!, force: Boolean): String
}`,
      options: [
        {
          bannedParameters: [{ keywords: ["force"], description: "42" }],
        },
      ],
      errors: 1,
    },
    {
      code: `
extend type Query {
  messageOfTheDay(date: Date!, cacheBuster: Boolean): String
}`,
      options: [
        {
          bannedParameters: [
            { keywords: ["cache"], partialMatch: true, description: "42" },
          ],
        },
      ],
      errors: 1,
    },
    {
      code: `
extend type Query {
  messageOfTheDay(date: Date!, noCache: Boolean): String
}`,
      options: [
        {
          bannedParameters: [
            { keywords: ["cache"], partialMatch: true, description: "42" },
          ],
        },
      ],
      errors: 1,
    },
    {
      code: `
extend type Query {
  messageOfTheDay(date: Date!, cacheBuster: Boolean, force: Boolean): String
}`,
      options: [
        {
          bannedParameters: [
            { keywords: ["cache"], partialMatch: true, description: "42" },
            {
              keywords: ["force"],
              description: "don't do that.",
            },
          ],
        },
      ],
      errors: 2,
    },
  ],
});
