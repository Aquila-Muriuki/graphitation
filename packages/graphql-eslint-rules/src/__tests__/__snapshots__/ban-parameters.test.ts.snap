// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | extend type Query {
      2 |   messageOfTheDay(date: Date!, force: Boolean): String
      3 | }

#### ⚙️ Options

    {
      "bannedParameters": [
        {
          "keywords": [
            "force"
          ],
          "description": "42"
        }
      ]
    }

#### ❌ Error

      1 | extend type Query {
    > 2 |   messageOfTheDay(date: Date!, force: Boolean): String
        |                                ^^^^^^^ "force" is a prohibited parameter name. Reason: 42"
      3 | }"
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

      1 | extend type Query {
      2 |   messageOfTheDay(date: Date!, cacheBuster: Boolean): String
      3 | }

#### ⚙️ Options

    {
      "bannedParameters": [
        {
          "keywords": [
            "cache"
          ],
          "partialMatch": true,
          "description": "42"
        }
      ]
    }

#### ❌ Error

      1 | extend type Query {
    > 2 |   messageOfTheDay(date: Date!, cacheBuster: Boolean): String
        |                                ^^^^^^^^^^^^^ "Parameter name can not contain "cache". Reason: 42"
      3 | }"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

      1 | extend type Query {
      2 |   messageOfTheDay(date: Date!, noCache: Boolean): String
      3 | }

#### ⚙️ Options

    {
      "bannedParameters": [
        {
          "keywords": [
            "cache"
          ],
          "partialMatch": true,
          "description": "42"
        }
      ]
    }

#### ❌ Error

      1 | extend type Query {
    > 2 |   messageOfTheDay(date: Date!, noCache: Boolean): String
        |                                ^^^^^^^^^ "Parameter name can not contain "cache". Reason: 42"
      3 | }"
`;

exports[`Invalid #4 1`] = `
"#### ⌨️ Code

      1 | extend type Query {
      2 |   messageOfTheDay(date: Date!, cacheBuster: Boolean, force: Boolean): String
      3 | }

#### ⚙️ Options

    {
      "bannedParameters": [
        {
          "keywords": [
            "cache"
          ],
          "partialMatch": true,
          "description": "42"
        },
        {
          "keywords": [
            "force"
          ],
          "description": "don't do that."
        }
      ]
    }

#### ❌ Error 1/2

      1 | extend type Query {
    > 2 |   messageOfTheDay(date: Date!, cacheBuster: Boolean, force: Boolean): String
        |                                ^^^^^^^^^^^^^ "Parameter name can not contain "cache". Reason: 42"
      3 | }

#### ❌ Error 2/2

      1 | extend type Query {
    > 2 |   messageOfTheDay(date: Date!, cacheBuster: Boolean, force: Boolean): String
        |                                                      ^^^^^^^ "force" is a prohibited parameter name. Reason: don't do that."
      3 | }"
`;
